### PR TITLE
Hub link

### PR DESCRIPTION
**Pull request by Xodarap**
I think this is mostly good, but want to wait another day to see if people have more suggestions about the text to use here

### CODE REVIEW

**Review by colophonemes — COMMENTED** [review]
R

File: packages/lesswrong/components/common/HashLink.tsx [hunk 1]
```diff
@@ -84,7 +85,14 @@ export function genericHashLink(props, As) {
 }
 
 export function HashLink(props) {
-  return genericHashLink(props, Link);
+  // React router links don't handle external URLs, so use a
+  // normal HTML a tag if the URL is external
+  const externalLink = props.to.slice(0,8) === 'https://'  
+  const Element = externalLink ? 
+    (p) => React.createElement('a', {...p, href: p.to})
```
Comment (colophonemes) [inline packages/lesswrong/components/common/HashLink.tsx:92]: `React.createElement` isn't really idiomatic. I'd rewrite this in plain JSX:

```js
({to, ...rest}) => <a href={to} {...rest} />
```

File: packages/lesswrong/components/common/HashLink.tsx [hunk 2]
```diff
@@ -84,7 +85,14 @@ export function genericHashLink(props, As) {
 }
 
 export function HashLink(props) {
-  return genericHashLink(props, Link);
+  // React router links don't handle external URLs, so use a
+  // normal HTML a tag if the URL is external
+  const externalLink = props.to.slice(0,8) === 'https://'
```
Comment (colophonemes) [inline packages/lesswrong/components/common/HashLink.tsx:90]: This isn't robust to HTTP links. I generally do this with a regex:
```js
const externalLink = /https?:\/\//.test(props.to)
```

File: packages/lesswrong/components/common/HashLink.tsx (hunk 1 above)
Comment (colophonemes) [reply packages/lesswrong/components/common/HashLink.tsx:92]: You might also want to consider adding `target='_blank'` to external links to open them in a new tab (not sure). If you do, [make sure you also add `rel="noopener"`](https://web.dev/external-anchors-use-rel-noopener/)

**Review by colophonemes — APPROVED** [review]
ASTLAC

**Comment by jpaddison3** [comment]
@colophonemes I just noticed that the task was assigned to you in asana. Sorry about that, no reason this should have been yours

**Comment by jpaddison3** [comment]
image

Huh, That sure looks like I should be the default. @Xodarap, did you assign Sam as the reviewer?

**Comment by colophonemes** [comment]
I assigned me as the reviewer when I got the Asana task, and I checked that you had assigned it to me to ensure that it was deliberate. However, maybe that's just a consequence of Zapier using your API key for that Zap?

image

**Comment by jpaddison3** [comment]
It is. You're always the one assigning my parfit or aletheia ones.

**Comment by Xodarap** [comment]
upstream https://github.com/LessWrong2/Lesswrong2/pull/3645 was accepted. Closing this PR.

**Comment by jpaddison3** [comment]
You could still merge this PR in case we deploy again before merging LW

**Pull request closed without merging**